src/ethz_snow/snowing.py

```diff
@@ -0,0 +1,2227 @@
+        temperature (str): Model dimensionality.
+        configuration (str): Freezing configuration.
+        plotting (bool): Whether evolutions are plotted or not.
+        Nrep (int): Number of repetitions.
+        configPath (Optional[str]): The path of the (optional) custom config yaml.
+    """
+
+    def __init__(
+        self,
+        k: dict = {"int": 0, "ext": 0, "s0": 50, "s_sigma_rel": 0},
+        opcond: OperatingConditions = OperatingConditions(),
+        temperature: str = "spatial_1D",
+        configuration: str = "shelf",
+        plotting: bool = True,
+        Nrep: int = 1,
+        configPath: Optional[str] = None,
+    ):
+        """Construct a Snowing object.
+
+        Args:
+            k (dict, optional): Heat transfer coefficients. Defaults to {"int": 0, "ext": 0, "s0": 50, "s_sigma_rel": 0}.
+            opcond (OperatingConditions, optional): Operating conditions of the run. Defaults to OperatingConditions().
+            temperature (str, optional): Model dimensionality. Defaults to None.
+            configuration (str, optional): Freezing configuration. Defaults to None.
+            plotting (bool, optional): Whether evolutions are plotted or not. Defaults to None.
+            Nrep (int, optional): Number of repetitions. Defaults to 1.
+            configPath (Optional[str], optional): The path of the (optional) custom config yaml. Defaults to None.
+
+        Raises:
+            TypeError: If k is not a dict.
+            ValueError: If a aspecific key is missing in dict k.
+            NotImplementedError: If freezing configuration is not correctly specified.
+            NotImplementedError: If model dimensionality is not correctly specified.
+            TypeError: If opcond is not of type operatingConditions.
+
+        Examples:
+            >>> S = Snowing()
+            >>> S = Snowing(k=d, opcond=op)
+            >>> S = Snowing(k=d, opcond=op, temperature="spatial_1D", configuration="shelf", plotting=True)
+            >>> S = Snowing(k=d, opcond=op, temperature="spatial_1D", configuration="shelf", plotting=False))
+            >>> S = Snowing(k=d, opcond=op, temperature="spatial_1D", configuration="visf", plotting=True))
+            >>> S = Snowing(k=d, opcond=op, temperature="spatial_2D", configuration="jacket"))
+        """
+
+        if not isinstance(k, dict):
+            raise TypeError(f"Input k must be of type dict. Was given {type(k)}.")
+        elif not all([key in k.keys() for key in HEATFLOW_REQUIREDKEYS]):
+            raise ValueError(
+                (
+                    f"A required key was missing from dictionary k, specifically "
+                    + f"{set(HEATFLOW_REQUIREDKEYS) - set(k.keys())}."
+                )
+            )
+        else:
+            self.k = k
+
+        # for saving the results
+        self._domain = None
+        self._prop = None
+        self._time = None
+        self._shelf = None
+        self._temp = None
```

Comment:
For consistency, it would be nice, albeit not critical, to use the same logic with a state variable X that includes both temp and ice, then make temp and ice properties that are derived from X

Reply:
I see, but I would leave it like this actually. I find it more clear